receiver/prometheusreceiver/config.go

```diff
@@ -165,8 +165,8 @@ func (cfg *Config) Validate() error {
 }
 
 func (cfg *Config) validatePromConfig(promConfig *promconfig.Config) error {
-	if len(promConfig.ScrapeConfigs) == 0 {
-		return errors.New("no Prometheus scrape_configs")
+	if len(promConfig.ScrapeConfigs) == 0 && cfg.TargetAllocator == nil {
```

Comment:
Wondering why is an empty scrape config valid if `cfg.TargetAllocator != nil`?

Reply:
We want to allow someone to specify only a target allocator to pull its configuration from rather than a target allocator and a job with some dummy data so that the collector can be run. 

Better said – some collectors may only want to use configuration from the target allocator's scrape_configs endpoint and nothing else.